app/services/count_delegates.rb

```diff
@@ -0,0 +1,39 @@
+class CountDelegates
+  def initialize
+    @primary_users =
+      User.primary_svie_member
+          .includes(primary_membership: [:posts, :post_types, { user: [:primary_membership] }])
+          .select { |user| user.primary_membership&.primary? }
+  end
+
+  def call
+    reset_delegate_count
+    count_delegates
+  end
+
+  private
+
+  def reset_delegate_count
+    Group.update_all grp_svie_delegate_nr: 0
+  end
+
+  def count_delegates
+    group_member_count = Hash.new 0
+    @primary_users.map { |user| user.primary_membership.group_id }
+                  .each { |group_id| group_member_count[group_id] += 1 }
+    group_member_count.each { |group_id, member_count| set_delegate_count(group_id, member_count) }
+  end
+
+  def set_delegate_count(group_id, member_count)
+    return if member_count < 5
+
+    group = Group.find(group_id)
```

Comment:
In `count_delegates` method I've only used `group_id`-s to avoid unnecessary memory allocation and preloading. There are some groups that have primary members, but less than five, so these groups aren't needed here.

Reply:
I stick to my statement.

And I remember that we decided that the database should contain up-to-date information.
So I think the `primary_svie_members` scope have to give the correct users. I mean if the DB not up-to-date at now, the logic should have to be in the scope.